qcodes/instrument_drivers/Keysight/KeysightAgilent_33XXX.py

```diff
@@ -0,0 +1,349 @@
+from qcodes.instrument.channel import InstrumentChannel, ChannelList
+from qcodes.instrument.base import Instrument
+
+log = logging.getLogger(__name__)
+
+
+# This is to be the grand unified driver superclass for
+# The Keysight/Agilent/HP Waveform generators of series
+# 33200, 33500, and 33600
+
+
+class OutputChannel(InstrumentChannel):
+    """
+    Class to hold the output channel of a waveform generator
+    """
+    def __init__(self, parent: Instrument, name: str, channum: int) -> None:
+        """
+        Args:
+            parent: The instrument to which the channel is
+                attached.
+            name: The name of the channel
+            channum: The number of the channel in question (1-2)
+        """
+        super().__init__(parent, name)
+
+        def val_parser(parser: type, inputstring: str) -> str:
+            """
+            Parses return values from instrument. Meant to be used when a query
+            can return a meaningful finite number or a numeric representation
+            of infinity
+
+            Args:
+                parser: Either int or float, what to return in finite
+                    cases
+                inputstring: The raw return value
+            """
+
+            inputstring = inputstring.strip()
+
+            if float(inputstring) == 9.9e37:
+                output = float('inf')
+            else:
+                output = float(inputstring)
+                if parser == int:
+                    output = parser(output)
+
+            return output
+
+        self.add_parameter('function_type',
+                           label='Channel {} function type'.format(channum),
+                           set_cmd='SOURce{}:FUNCtion {{}}'.format(channum),
+                           get_cmd='SOURce{}:FUNCtion?'.format(channum),
+                           get_parser=str.rstrip,
+                           vals=vals.Enum('SIN', 'SQU', 'TRI', 'RAMP',
+                                          'PULS', 'PRBS', 'NOIS', 'ARB',
+                                          'DC')
+                           )
+
+        self.add_parameter('frequency_mode',
+                           label='Channel {} frequency mode'.format(channum),
+                           set_cmd='SOURce{}:FREQuency:MODE {{}}'.format(channum),
+                           get_cmd='SOURce{}:FREQuency:MODE?'.format(channum),
+                           get_parser=str.rstrip,
+                           vals=vals.Enum('CW', 'LIST', 'SWEEP', 'FIXED')
+                           )
+
+        self.add_parameter('frequency',
+                           label='Channel {} frequency'.format(channum),
+                           set_cmd='SOURce{}:FREQuency {{}}'.format(channum),
+                           get_cmd='SOURce{}:FREQuency?'.format(channum),
+                           get_parser=float,
+                           unit='Hz',
+                           # TODO: max. freq. actually really tricky
```

Comment:
Does this need to depend on the model? I don't know why it's really tricky (does it perhaps depend on output type?)

Reply:
It depends on model (I just pushed that change), but also other things which I don't remember. Amplitude, prolly.

Reply:
never mind